project/CodegenPlugin.scala

```diff
@@ -166,6 +213,10 @@ object CodegenPlugin extends AutoPlugin {
         version.value
       }
     },
+    dotnetVersion := {
+      val versionArray = version.value.split("-".toCharArray)
+      versionArray.head + "-rc" + versionArray.drop(1).dropRight(1).mkString("")
+    },
```

Comment:
Im so sorry to keep bugging you on this but i still think thise code might have some suprises. In particular i think you'll want to check whether the version is a release, snapshot, or dirty snapshot and combine accordingly. I think as it stands this will error on a release version because it wont have any "-" characters. I think its ok for dirty though because of your drop and dropRight calls.
```suggestion
    dotnetVersion := {
      if (version.value.contains("-") ){
        val versionArray = version.value.split("-".toCharArray)
        versionArray.head + "-rc" + versionArray.drop(1).dropRight(1).mkString("")
      } else {
       version.value
      }
    },
```